aws-rds-cfn-common/src/main/java/software/amazon/rds/common/handler/LoggerWrapper.java

```diff
@@ -0,0 +1,73 @@
+package software.amazon.rds.common.handler;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+@lombok.Getter
+@lombok.Setter
+public class LoggerWrapper {
+
+    private Logger logger;
+    private String clientRequestToken;
+    private String awsAccountId;
+    private String stackId;
+
+    public <T> LoggerWrapper(Logger logger, ResourceHandlerRequest<T> request) {
+        this.logger = logger;
+        if (request != null) {
+            clientRequestToken = request.getClientRequestToken();
+            awsAccountId = request.getAwsAccountId();
+            stackId = request.getStackId();
+        }
+    }
+
+
+    public void logThrowableAndRethrow(Throwable throwable) {
+        try {
+            StringBuilder sb = new StringBuilder();
+            sb.append(ExceptionUtils.getStackTrace(throwable));
+            addCustomerData(sb);
+            logMessage(sb.toString());
+        } catch (Throwable caughtThrowable) {
+            logMessage(caughtThrowable.toString());
+        }
+        ExceptionUtils.rethrow(throwable);
+    }
+
+    public void log(String message, Object object) {
+        log(message, object, Collections.emptyList());
+    }
+
+    //Adding data at the end for better experience. To make first line in log message meaningful 
+    public void log(String message, Object object, Collection<String> excludeParameters) {
```

Comment:
I believe the "exclude parameters" is somewhat an open-by-default approach, meaning that we can make a mistake and leak some data that would rather be omitted (compare this to "closed by default", requiring one to be explicit what is **enabled** instead). I believe we can take it even further and aim for a "safe by default" approach. One way to achieve it would be to:
- define a `LogDecorator` interface and ensure our overloaded logger always passes the log data through it
- define a `ObfuscateLogDecorator` class, defining the range of attributes that need to be obfuscated or replaced.
- The list of decorators could be passed along upon a logger instantiation and hence won't be required when a regular `log()` is being invoked.